OvmfPkg/OvmfPkgIa32X64.dsc

```diff
@@ -252,6 +253,20 @@
 !include OvmfPkg/Include/Dsc/OvmfTpmLibs.dsc.inc
 !include OvmfPkg/Include/Dsc/ShellLibs.dsc.inc
 
+[LibraryClasses.common.SEC, LibraryClasses.common.PEI_CORE]
+  # don't check stack cookies in SEC and PEI_CORE because we don't have exception handlers installed at this point and
+  # will triple fault and reboot if we hit a stack cookie failure
+  NULL|MdePkg/Library/StackCheckLibNull/StackCheckLibNull.inf
+
+[LibraryClasses.common.PEIM]
+  # because some PEIMs run pre-memory, globally have all PEIMs use the static stack cookie
+  NULL|MdePkg/Library/StackCheckLib/StackCheckLib.inf
+
+[LibraryClasses.common.DXE_CORE, LibraryClasses.common.SMM_CORE, LibraryClasses.common.DXE_SMM_DRIVER, LibraryClasses.common.DXE_DRIVER, LibraryClasses.common.DXE_RUNTIME_DRIVER, LibraryClasses.common.UEFI_DRIVER, LibraryClasses.common.UEFI_APPLICATION]
+  # for all other types, use the dynamic stack cookie
+  NULL|MdePkg/Library/StackCheckLib/StackCheckLib.inf
+  StackCheckEntryPointLib|MdePkg/Library/StackCheckLib/StackCheckLibDynamicInit.inf
+

```

Comment:
Place this in a include file in `OvmfPkg/Include/Dsc/` ?
That way it can be easily added to more builds without cut+pasting this into all the *.dsc files.

Reply:
Sure, I'll move this to a dsc.inc.